### PR TITLE
Changes to create breaks in time series charts when there are gaps in the weeks and other small fixes.

### PR DESCRIPTION
**Pull request by RalphMorita**

### CODE REVIEW

File: src/TimeSeries.svelte [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
   import * as d3 from "d3";
   import { _ } from "svelte-i18n";
+import { addScalarDependencies } from "mathjs";
```
Comment (mimisun) [inline src/TimeSeries.svelte:36]: indent

File: src/choropleth.ts [hunk 2]
```diff
@@ -844,6 +844,14 @@ function drawMapCalloutInfo(data, fipsCode) {
       sni_safety_side_effects: 0.0,
     };
   }
+  // potential fix for map issue that doesn't show "not enough data" message for NaNs
+  else if (typeof trends != "undefined" && isNaN(trends.sni_covid19_vaccination) && isNaN(trends.sni_safety_side_effects) && isNaN(trends.sni_vaccination_intent)) {
```
Comment (mimisun) [inline src/choropleth.ts:848]: merge this with the if block above.

File: src/stores.ts [hunk 3]
```diff
@@ -72,3 +74,27 @@ export const regionalTrends = writable(rt);
...
 
+/** 
+ * A store to contain an array of all valid dates
+ */
+export const dateRange = writable<string[]>(fetchDateData());
+/**
+ * A method to read in the ALL_dates.csv file into an array of dates strings.
+ * @returns an array of date string values
+ */
+ function fetchDateData() {
```
Comment (mimisun) [inline src/stores.ts:85]: data fetching and parsing functions should live in data.ts.

File: src/TimeSeries.svelte [hunk 4]
```diff
@@ -490,6 +492,111 @@
       }
     });
 
+    // add NaNs to the regional data in order to make the time series "break" on gaps in weeks.
+    addNans(data)
+
+    /*  A function to add NaNs to the data in order to force breaks in the time series
```
Comment (mimisun) [inline src/TimeSeries.svelte:498]: remove all commented out code.

File: src/TimeSeries.svelte [hunk 5]
```diff
@@ -490,6 +492,111 @@
...
+          let missingDates = new Set([...allDatesSet].filter(x => !dateSet.has(x)));
+          //console.log(missingDates);
+          missingDates.forEach((missingDate) => {
+            datesToAdd.push({
+              date: missingDate,
+              value: NaN
+            })
+          });
+          trendVal.push(...datesToAdd)
+          trendVal.sort((a: TrendValue,b: TrendValue) => new Date(a.date).getTime() - new Date(b.date).getTime());
```
Comment (mimisun) [inline src/TimeSeries.svelte:592]: You can compare the strings without any conversions.
We use the YYYY-mm-dd format so it's sortable as strings.

**Review by mimisun — COMMENTED** [review]
LGTM overall, just some small changes needed.

File: src/TimeSeries.svelte (hunk 4 above)
Comment (RalphMorita) [reply src/TimeSeries.svelte]: done

File: src/choropleth.ts (hunk 2 above)
Comment (RalphMorita) [reply src/choropleth.ts]: merged.

File: src/stores.ts (hunk 3 above)
Comment (RalphMorita) [reply src/stores.ts]: moved this block of code to data.ts.

File: src/TimeSeries.svelte (hunk 5 above)
Comment (RalphMorita) [reply src/TimeSeries.svelte]: using string comparison. let me know if this comparison function is ok.

**Review by mimisun — APPROVED** [review]
Looks a lot cleaner, thank you!

**Pull request merged**